compiler/compile/OMRCompilation.cpp

```diff
@@ -294,9 +295,17 @@ OMR::Compilation::Compilation(
    _gpuPtxCount(0),
    _bitVectorPool(self()),
    _typeLayoutMap((LayoutComparator()), LayoutAllocator(self()->region())),
-   _target(TR::Compiler->target),
    _tlsManager(*self())
    {
+   if (target != NULL)
```

Comment:
I don't see anything wrong with the code. I'm just wondering if we should force the Compilation object to take a target every time so that we don't have to have a default value ("TR::Compiler->target") for it.

Reply:
If the default value will almost always be `TR::Compiler` anyway, then I don't think we should burden the user with having to specify it. There could be many uses of this constructor beyond OpenJ9's JITServer/AOT at some point. So I went in this direction because I figured for the most part users of the API shouldn't have to worry about specifying this field, unless they want to send in a customized target. They didn't in the past. So the goal here was to try to keep the original behaviour as much as possible.

Reply:
We do have a problem with Portable AOT though. The default value will not always be `TR::Compiler->target` for an AOT compilation.

Reply:
Sorry I don't understand the issue here. If it won't be `TR::Compiler->target`, then won't we be able to specify whatever it is with this new API change?

Reply:
To summarize an offline discussion with @harryyu1994 : We won't have a problem with Portable AOT because the only use of the `TR::Compilation` constructor in OpenJ9 is inside wrapped compile, and we handle the portable aot case inside there. See https://github.com/eclipse/openj9/pull/10474 for the change.